buster/retriever/deeplake.py

```diff
@@ -9,17 +11,67 @@
 logging.basicConfig(level=logging.INFO)
 
 
+def extract_metadata(x, columns):
+    """Returned metadata from deeplake is in a nested dict, extract it so that each attribute has its own column."""
+    for col in columns:
+        x[col] = x.metadata[col]
+    return x
+
+
+def data_dict_to_df(data: dict):
+    # rename 'score' to 'similarity'
+    data["similarity"] = data.pop("score")
+    data["content"] = data.pop("text")
+
+    matched_documents = pd.DataFrame(data)
+
+    if len(matched_documents) == 0:
+        logger.info("No matches found...")
+        return pd.DataFrame()
+
+    matched_documents = matched_documents.apply(extract_metadata, columns=["source", "title", "url"], axis=1)
+    matched_documents = matched_documents.drop(columns="metadata")
+
+    return matched_documents
+
+
+def build_tql_query(embedding, sources=None, top_k: int = 3):
+    # Initialize the where_clause to an empty string.
+    where_clause = ""
+
+    embedding_string = ",".join([str(item) for item in embedding])
+
+    # If sources is provided and it's not empty, build the where clause.
+    if sources:
+        conditions = [f"contains(metadata['source'], '{source}')" for source in sources]
+        where_clause = "where " + " or ".join(conditions)
+
+    # Construct the entire query
+    query = f"""
+select * from (
+    select embedding, text, metadata, cosine_similarity(embedding, ARRAY[{embedding_string}]) as score
+    {where_clause}
+)
+order by score desc limit {top_k}
+"""
+    return query
+
+
 class DeepLakeRetriever(Retriever):
-    def __init__(self, path, **kwargs):
+    def __init__(self, path, exec_option: str = "python", org_id: str = None, use_tql: bool = False, **kwargs):
         from deeplake.core.vectorstore import VectorStore
 
         super().__init__(**kwargs)
+        self.use_tql = use_tql
+        self.exec_option = exec_option
         self.vector_store = VectorStore(
             path=path,
             read_only=True,
+            token=os.getenv("ACTIVELOOP_TOKEN"),
```

Comment:
should be exposed in the init and loaded elsewhere

Reply:
done, added a warning if the variable isn't set (you should still be able to use the library just fine without it if not using enterprise features)